hs_collection_resource/tests/test_collection.py

```diff
@@ -0,0 +1,414 @@
+                      ]
+        )
+
+        self.resGen1 = create_resource(
+            resource_type='GenericResource',
+            owner=self.user1,
+            title='Gen 1',
+            keywords=['kw1', 'kw2'],
+            metadata=[{"rights": {"statement": "mystatement", "url": "http://www.google.com"}},
+                  {"description": {"abstract": "myabstract"}}
+                  ]
+        )
+
+        self.resGen2 = create_resource(
+            resource_type='GenericResource',
+            owner=self.user1,
+            title='Gen 2',
+            keywords=['kw1', 'kw2'],
+            metadata=[{"rights": {"statement": "mystatement", "url": "http://www.google.com"}},
+            {"description": {"abstract": "myabstract"}}
+            ]
+        )
+
+        self.resGen3 = create_resource(
+            resource_type='GenericResource',
+            owner=self.user1,
+            title='Gen 3',
+            keywords=['kw1', 'kw2'],
+            metadata=[{"rights": {"statement": "mystatement", "url": "http://www.google.com"}},
+            {"description": {"abstract": "myabstract"}}
+            ]
+        )
+
+        self.user2 = create_account(
+            'byu2@byu.edu',
+            username='user2',
+            password='mypassword2',
+            first_name='myfirstname2',
+            last_name='mylastname2',
+            superuser=False,
+            groups=[self.group]
+        )
+
+        self.resGen4 = create_resource(
+            resource_type='GenericResource',
+            owner=self.user2,
+            title='Gen 4',
+            keywords=['kw1', 'kw2'],
+            metadata=[{"rights": {"statement": "mystatement", "url": "http://www.google.com"}},
+            {"description": {"abstract": "myabstract"}}
+            ]
+        )
+
+        self.url_to_update_collection = "/hsapi/_internal/{0}/update-collection/"
+        self.url_to_collection_member_permission = "/hsapi/_internal/{0}/collection-member-permission/{1}/"
+        self.url_to_set_resource_flag = "/hsapi/_internal/{0}/set-resource-flag/"
+        self.url_to_delete_resource = "/hsapi/_internal/{0}/delete-resource/"
+
+    def tearDown(self):
+        super(TestCollection, self).tearDown()
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def test_views(self):
+        # test update_collection()
+        # self.assertEqual(self.resCollection.metadata.collection.all().count(), 0)
+        self.assertEqual(self.resCollection.metadata.collection, None)
+        url_to_update_collection = self.url_to_update_collection.format(self.resCollection.short_id)
+
+        # anonymous user
+        # should inform frontend error
+        response = self.api_client.post(url_to_update_collection,
+                                        {'resource_id_list': [self.resGen1.short_id, self.resGen2.short_id, self.resGen3.short_id]},
+                                        HTTP_REFERER='http://foo/bar')
```

Comment:
@pkdash I think the answer is yes, because this view function calls other view functions in hs_core and those view functions needs request.META['HTTP_REFERER'].